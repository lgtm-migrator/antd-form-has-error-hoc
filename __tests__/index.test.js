import React from 'react'
import { shallow, mount } from 'enzyme'
import { Form, Input } from 'antd'
import toJson from 'enzyme-to-json'

import withAntdFormHasError from '../src'

describe('antd form create hoc test', () => {
  @Form.create()
  @withAntdFormHasError()
  class Test extends React.Component {
    render() {
      const { getFieldDecorator } = this.props.form
      return (
        <Form>
          <Form.Item>
            {getFieldDecorator('username', {
              rules: [
                { required: true, message: 'Please input your username!' }
              ]
            })(<Input placeholder="Username" />)}
          </Form.Item>
          <Form.Item>
            {getFieldDecorator('password', {
              rules: [
                { required: true, message: 'Please input your Password!' }
              ]
            })(<Input type="password" placeholder="Password" />)}
          </Form.Item>
        </Form>
      )
    }
  }
  it('should get hasError from this.props', () => {
    const wrapper = mount(<Test />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('should get hasError from this.props', () => {
    const wrapper = shallow(<Test />)
    expect(wrapper.props().hasError).toEqual(true)
  })
})
