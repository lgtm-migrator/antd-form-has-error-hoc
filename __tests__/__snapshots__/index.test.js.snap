// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`antd form create hoc test should get hasError from this.props 1`] = `
<Form(WrappedComponent)>
  <ForwardRef
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
  >
    <AntdFormHasError(Test)
      form={
        Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        }
      }
      forwardRef={null}
    >
      <Test
        form={
          Object {
            "getFieldDecorator": [Function],
            "getFieldError": [Function],
            "getFieldInstance": [Function],
            "getFieldProps": [Function],
            "getFieldValue": [Function],
            "getFieldsError": [Function],
            "getFieldsValue": [Function],
            "isFieldTouched": [Function],
            "isFieldValidating": [Function],
            "isFieldsTouched": [Function],
            "isFieldsValidating": [Function],
            "isSubmitting": [Function],
            "resetFields": [Function],
            "setFields": [Function],
            "setFieldsInitialValue": [Function],
            "setFieldsValue": [Function],
            "submit": [Function],
            "validateFields": [Function],
            "validateFieldsAndScroll": [Function],
          }
        }
        hasError={true}
        updateFieldsStatus={[Function]}
      >
        <Form
          colon={true}
          hideRequiredMark={false}
          layout="horizontal"
          onSubmit={[Function]}
        >
          <form
            className="ant-form ant-form-horizontal"
            onSubmit={[Function]}
          >
            <FormItem
              hasFeedback={false}
            >
              <Row
                className="ant-form-item"
                gutter={0}
                key="row"
              >
                <div
                  className="ant-row ant-form-item"
                >
                  <Col
                    className="ant-form-item-control-wrapper"
                  >
                    <div
                      className="ant-col ant-form-item-control-wrapper"
                    >
                      <div
                        className="ant-form-item-control"
                      >
                        <span
                          className="ant-form-item-children"
                        >
                          <Input
                            data-__field={
                              Object {
                                "errors": null,
                                "name": "username",
                                "status": null,
                                "touched": false,
                                "value": undefined,
                              }
                            }
                            data-__meta={
                              Object {
                                "name": "username",
                                "originalProps": Object {
                                  "placeholder": "Username",
                                  "type": "text",
                                },
                                "ref": null,
                                "rules": Array [
                                  Object {
                                    "message": "Please input your username!",
                                    "required": true,
                                  },
                                ],
                                "trigger": "onChange",
                                "validate": Array [
                                  Object {
                                    "rules": Array [
                                      Object {
                                        "message": "Please input your username!",
                                        "required": true,
                                      },
                                    ],
                                    "trigger": Array [
                                      "onChange",
                                    ],
                                  },
                                ],
                                "valuePropName": "value",
                              }
                            }
                            id="username"
                            onChange={[Function]}
                            placeholder="Username"
                            type="text"
                          >
                            <input
                              className="ant-input"
                              data-__field={
                                Object {
                                  "errors": null,
                                  "name": "username",
                                  "status": null,
                                  "touched": false,
                                  "value": undefined,
                                }
                              }
                              data-__meta={
                                Object {
                                  "name": "username",
                                  "originalProps": Object {
                                    "placeholder": "Username",
                                    "type": "text",
                                  },
                                  "ref": null,
                                  "rules": Array [
                                    Object {
                                      "message": "Please input your username!",
                                      "required": true,
                                    },
                                  ],
                                  "trigger": "onChange",
                                  "validate": Array [
                                    Object {
                                      "rules": Array [
                                        Object {
                                          "message": "Please input your username!",
                                          "required": true,
                                        },
                                      ],
                                      "trigger": Array [
                                        "onChange",
                                      ],
                                    },
                                  ],
                                  "valuePropName": "value",
                                }
                              }
                              id="username"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Username"
                              type="text"
                              value=""
                            />
                          </Input>
                        </span>
                        <Animate
                          animation={Object {}}
                          component=""
                          componentProps={Object {}}
                          key="help"
                          onAppear={[Function]}
                          onEnd={[Function]}
                          onEnter={[Function]}
                          onLeave={[Function]}
                          transitionAppear={true}
                          transitionEnter={true}
                          transitionLeave={true}
                          transitionName="show-help"
                        />
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </FormItem>
            <FormItem
              hasFeedback={false}
            >
              <Row
                className="ant-form-item"
                gutter={0}
                key="row"
              >
                <div
                  className="ant-row ant-form-item"
                >
                  <Col
                    className="ant-form-item-control-wrapper"
                  >
                    <div
                      className="ant-col ant-form-item-control-wrapper"
                    >
                      <div
                        className="ant-form-item-control"
                      >
                        <span
                          className="ant-form-item-children"
                        >
                          <Input
                            data-__field={
                              Object {
                                "errors": null,
                                "name": "password",
                                "status": null,
                                "touched": false,
                                "value": undefined,
                              }
                            }
                            data-__meta={
                              Object {
                                "name": "password",
                                "originalProps": Object {
                                  "placeholder": "Password",
                                  "type": "password",
                                },
                                "ref": null,
                                "rules": Array [
                                  Object {
                                    "message": "Please input your Password!",
                                    "required": true,
                                  },
                                ],
                                "trigger": "onChange",
                                "validate": Array [
                                  Object {
                                    "rules": Array [
                                      Object {
                                        "message": "Please input your Password!",
                                        "required": true,
                                      },
                                    ],
                                    "trigger": Array [
                                      "onChange",
                                    ],
                                  },
                                ],
                                "valuePropName": "value",
                              }
                            }
                            id="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                          >
                            <input
                              className="ant-input"
                              data-__field={
                                Object {
                                  "errors": null,
                                  "name": "password",
                                  "status": null,
                                  "touched": false,
                                  "value": undefined,
                                }
                              }
                              data-__meta={
                                Object {
                                  "name": "password",
                                  "originalProps": Object {
                                    "placeholder": "Password",
                                    "type": "password",
                                  },
                                  "ref": null,
                                  "rules": Array [
                                    Object {
                                      "message": "Please input your Password!",
                                      "required": true,
                                    },
                                  ],
                                  "trigger": "onChange",
                                  "validate": Array [
                                    Object {
                                      "rules": Array [
                                        Object {
                                          "message": "Please input your Password!",
                                          "required": true,
                                        },
                                      ],
                                      "trigger": Array [
                                        "onChange",
                                      ],
                                    },
                                  ],
                                  "valuePropName": "value",
                                }
                              }
                              id="password"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Password"
                              type="password"
                              value=""
                            />
                          </Input>
                        </span>
                        <Animate
                          animation={Object {}}
                          component=""
                          componentProps={Object {}}
                          key="help"
                          onAppear={[Function]}
                          onEnd={[Function]}
                          onEnter={[Function]}
                          onLeave={[Function]}
                          transitionAppear={true}
                          transitionEnter={true}
                          transitionLeave={true}
                          transitionName="show-help"
                        />
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </FormItem>
          </form>
        </Form>
      </Test>
    </AntdFormHasError(Test)>
  </ForwardRef>
</Form(WrappedComponent)>
`;
